### ISSUE TITLE
Realtime event broadcasting through REST calls does not work

### ISSUE THREAD

**Issue by FranciscoMessina** [supabase/realtime#774]
**Describe the bug**
When attempting to send a Realtime broadcast without establishing a Web Socket connection from a backend Node application, using the documented `channel.send()` method, it returns the string  `error` without any explanation.

**To Reproduce**
1. Have a supabase project running locally
2. Attempt to send a broadcast with the following code 
```typescript
import { createClient } from '@supabase/supabase-js';
import { Env } from '../../config/env';


const supabaseClient = createClient(Env.SUPABASE_URL, Env.SUPABASE_SERVICE_KEY)

supabaseClient.channel('channel1').send({
   event: 'event1',
   type: 'broadcast',
   payload: {
      some: 'data'
   }
}).then(res => { console.log(res)})
```

4. You should see the string "error" as a result, and any connected clients listening to that event won't be notified

**Expected behavior**
 I would expect it to work or at least give an informative reason as to why it is not working.

**Desktop (please complete the following information):**
 - OS: PopOS (linux) and Windows
 - Version of CLI 1.127.4
 - Version of supabase-js 2.39.2
 - Version of Node.js (if applicable) 20.9.0

**Comment by filipecabaco** [comment]
Hi

would it be possible to provide some logs from the container `supabase/realtime` @FranciscoMessina ?

**Comment by FranciscoMessina** [comment]
Yes of course, these are the logs that appear when attempting to send the event.

```bash
2024-01-09 10:32:15 13:32:15.692 [debug] QUERY OK source="extensions" db=0.5ms idle=386.7ms
2024-01-09 10:32:15 SELECT e0."id", e0."type", e0."settings", e0."tenant_external_id", e0."inserted_at", e0."updated_at", e0."tenant_external_id" FROM "extensions" AS e0 WHERE (e0."tenant_external_id" = $1) ORDER BY e0."tenant_external_id" ["realtime-dev"]
2024-01-09 10:32:15 13:32:15.693 project=realtime-dev [info] Billing metrics: [:realtime, :connections]
2024-01-09 10:32:20 13:32:20.699 project=realtime-dev [info] Billing metrics: [:realtime, :connections]
2024-01-09 10:32:20 13:32:20.775 request_id=F6ixD-R0MPRfEjMAAAJB [info] POST /socket/api/broadcast
2024-01-09 10:32:20 13:32:20.775 request_id=F6ixD-R0MPRfEjMAAAJB [info] Sent 404 in 360µs
2024-01-09 10:32:20 13:32:20.775 request_id=F6ixD-R0MPRfEjMAAAJB [debug] Converted error Phoenix.Router.NoRouteError to 404 response
2024-01-09 10:32:25 13:32:25.702 project=realtime-dev [info] Billing metrics: [:realtime, :connections]
```

**Comment by filipecabaco** [comment]
@w3b6x9 could it be on the client? it's weird that it has a request going to `/socket/api/broadcast` 👀

**Comment by Mukhammadali** [comment]
@filipecabaco 
I am also facing the same issue, `channel.send` returns `error` string, then I intercepted the network request and noticed that 
`https://<ID>.supabase.co/realtime/v1/api/broadcast` is returning 401 with the following error. 

<img width="660" alt="image" src="https://github.com/supabase/realtime/assets/22298895/b9fb64ad-1ee9-40bb-b62a-68a465475f91">

Library version: `@supabase/supabase-js: ^2.39.2`
Environment: React Native (Expo)

I followed the supabase documentation, everything is working fine including realtime websocket connection. WSS  is connected with the same apikey and access token and it is receiving events. No issues at all.
The only issue is with `channel.send` which is using REST api instead of websockets and that returns 401

**Comment by Mukhammadali** [comment]
The issue seems to be [here in this line](https://github.com/supabase/realtime-js/blob/3c5c3db038e14c52888a401db1bc142d567a9aa9/src/RealtimeChannel.ts#L439) in realtime-js package. `this.socket.accessToken` gives user accessToken not `anon` key, therefore that endpoint is returning 401

**Comment by Mukhammadali** [comment]
It was reported in realtime-js repo
https://github.com/supabase/realtime-js/issues/271

**Comment by w3b6x9** [comment]
this has been fixed in `realtime-js` v2.9.2. will get the new version in `supabase-js` soon but you can upgrade it yourself if you don't want to wait.

thanks everyone for the feedback!
